'MarshalInterface<IStringable>' does not contain a definition for 'CopyAbiArray'

Looks like this PR introduced the scenario we should probably be handling but aren't: https://github.com/microsoft/TestWinRT/pull/29/files specifically 

delegate Windows.Foundation.IStringable[] Array16Handler(Windows.Foundation.IStringable[] a, ref Windows.Foundation.IStringable[] b, out Windows.Foundation.IStringable[] c);



Trying to build the project gives the following error(s):

```
"C:\Users\ujchadha\source\repos\CsWinRT\cswinrt.sln" (default target) (1) ->
"C:\Users\ujchadha\source\repos\CsWinRT\UnitTest\UnitTest.csproj" (default target) (11) ->
"C:\Users\ujchadha\source\repos\CsWinRT\UnitTest\UnitTest.csproj" (Build target) (11:2) ->
"C:\Users\ujchadha\source\repos\CsWinRT\Projections\Test\Test.csproj" (default target) (14:2) ->
(CoreCompile target) ->
  Generated Files\TestComponent.cs(1380,78): error CS0117: 'MarshalInterface<IStringable>' does not contain a definition for 'CopyAbiArray' [C:\Users\
ujchadha\source\repos\CsWinRT\Projections\Test\Test.csproj]
  Generated Files\TestComponent.cs(7109,74): error CS0117: 'MarshalInterface<IStringable>' does not contain a definition for 'CopyAbiArray' [C:\Users\
ujchadha\source\repos\CsWinRT\Projections\Test\Test.csproj]


"C:\Users\ujchadha\source\repos\CsWinRT\cswinrt.sln" (default target) (1) ->
"C:\Users\ujchadha\source\repos\CsWinRT\UnitTest\UnitTest.csproj" (default target) (11) ->
"C:\Users\ujchadha\source\repos\CsWinRT\UnitTest\UnitTest.csproj" (Build target) (11:3) ->
"C:\Users\ujchadha\source\repos\CsWinRT\Projections\Test\Test.csproj" (default target) (14:4) ->
  Generated Files\TestComponent.cs(1380,78): error CS0117: 'MarshalInterface<IStringable>' does not contain a definition for 'CopyAbiArray' [C:\Users\
ujchadha\source\repos\CsWinRT\Projections\Test\Test.csproj]
  Generated Files\TestComponent.cs(7109,74): error CS0117: 'MarshalInterface<IStringable>' does not contain a definition for 'CopyAbiArray' [C:\Users\
ujchadha\source\repos\CsWinRT\Projections\Test\Test.csproj]
```

--- a/UnitTest/TestComponent_Tests.cs
+++ b/UnitTest/TestComponent_Tests.cs
@@ -421,6 +421,20 @@ namespace UnitTest
             Assert.True(AllEqual(a, b, c, d));
         }
 
+        [Fact]
+        public void Array_Stringable()
+        {
+            IStringable[] a = new IStringable[] {
+                Windows.Data.Json.JsonValue.CreateNumberValue(3),
+                Windows.Data.Json.JsonValue.CreateNumberValue(4),
+                Windows.Data.Json.JsonValue.CreateNumberValue(5.0)
+            };
+            IStringable[] b = new IStringable[a.Length];
+            IStringable[] c;
+            IStringable[] d = Tests.Array16(a, b, out c);
+            Assert.True(AllEqual(a, b, c, d));
+        }
+
         private T[] Array_Call<T>(T[] a, T[] b, out T[] c)
         {
             Assert.True(a.Length == b.Length);
@@ -517,6 +531,12 @@ namespace UnitTest
         public void Array_Nested_Call()
         {
             Tests.Array15Call(Array_Call);
+        }
+
+        [Fact]
+        public void Array_Stringable_Call()
+        {
+            Tests.Array16Call(Array_Call);
         }
 
         [Fact]

--- a/WinRT.Runtime/Marshalers.cs
+++ b/WinRT.Runtime/Marshalers.cs
@@ -685,6 +685,20 @@ namespace WinRT
             return array;
         }
 
+        public static unsafe void CopyAbiArray(T[] array, object box, Func<IntPtr, T> fromAbi)
+        {
+            if (box is null)
+            {
+                return;
+            }
+            var abi = ((int length, IntPtr data))box;
+            var data = (IntPtr*)abi.data.ToPointer();
+            for (int i = 0; i < abi.length; i++)
+            {
+                array[i] = fromAbi(data[i]);
+            }
+        }
+
         public static unsafe (int length, IntPtr data) FromManagedArray(T[] array, Func<T, IntPtr> fromManaged)
         {
             if (array is null)
@@ -866,6 +880,8 @@ namespace WinRT
         public static (int length, IntPtr data) GetAbiArray(object box) => MarshalInterfaceHelper<T>.GetAbiArray(box);
 
         public static unsafe T[] FromAbiArray(object box) => MarshalInterfaceHelper<T>.FromAbiArray(box, FromAbi);
+
+        public static unsafe void CopyAbiArray(T[] array, object box) => MarshalInterfaceHelper<T>.CopyAbiArray(array, box, FromAbi);
 
         public static unsafe (int length, IntPtr data) FromManagedArray(T[] array) => MarshalInterfaceHelper<T>.FromManagedArray(array, (o) => FromManaged(o));
 

--- a/WinRT.Runtime/Projections/Uri.cs
+++ b/WinRT.Runtime/Projections/Uri.cs
@@ -131,7 +131,7 @@ namespace ABI.System
                 return null;
             }
 
-            using var uri = ObjectReference<ABI.Windows.Foundation.IUriRuntimeClassVftbl>.FromAbi(ptr);
+            using var uri = ObjectReference<IUnknownVftbl>.FromAbi(ptr).As<ABI.Windows.Foundation.IUriRuntimeClassVftbl>();
             IntPtr rawUri = IntPtr.Zero;
             try
             {

--- a/get_testwinrt.cmd
+++ b/get_testwinrt.cmd
@@ -12,7 +12,7 @@ git checkout -f master
 if ErrorLevel 1 popd & exit /b !ErrorLevel!
 git fetch -f
 if ErrorLevel 1 popd & exit /b !ErrorLevel!
-git reset -q --hard 6d72afbcb51ab3981c6cd620d24954020f4d2bbc
+git reset -q --hard e058f2fc16e4812a16b5453da7ef4d65b3cf0bfe
 if ErrorLevel 1 popd & exit /b !ErrorLevel!
 echo Restoring Nuget
 ..\.nuget\nuget.exe restore

